kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/jms/JmsResponseHandlerIntegrationTest.java

```diff
@@ -0,0 +1,260 @@
+ * limitations under the License.
+*/
+
+package org.kie.server.integrationtests.jbpm.jms;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runners.Parameterized;
+import org.kie.server.api.marshalling.Marshaller;
+import org.kie.server.api.marshalling.MarshallerFactory;
+import org.kie.server.api.marshalling.MarshallingFormat;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.api.model.ServiceResponsesList;
+import org.kie.server.api.model.instance.ProcessInstance;
+import org.kie.server.api.model.instance.ProcessInstanceList;
+import org.kie.server.api.model.instance.TaskSummary;
+import org.kie.server.api.model.instance.TaskSummaryList;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.jms.AsyncResponseHandler;
+import org.kie.server.client.jms.BlockingResponseCallback;
+import org.kie.server.client.jms.FireAndForgetResponseHandler;
+import org.kie.server.client.jms.ReqReplyResponseHandler;
+import org.kie.server.client.jms.ResponseCallback;
+import org.kie.server.client.jms.ResponseHandler;
+import org.kie.server.integrationtests.category.JMSOnly;
+import org.kie.server.integrationtests.config.TestConfig;
+import org.kie.server.integrationtests.jbpm.JbpmKieServerBaseIntegrationTest;
+import org.kie.server.integrationtests.shared.KieServerAssert;
+import org.kie.server.integrationtests.shared.KieServerDeployer;
+
+import static org.junit.Assert.*;
+
+@Category({JMSOnly.class})
+public class JmsResponseHandlerIntegrationTest extends JbpmKieServerBaseIntegrationTest {
+
+    private static final ReleaseId RELEASE_ID = new ReleaseId("org.kie.server.testing", "definition-project", "1.0.0.Final");
+
+    @Parameterized.Parameters(name = "{index}: {0} {1}")
+    public static Collection<Object[]> data() {
+        KieServicesConfiguration jmsConfiguration = createKieServicesJmsConfiguration();
+        Collection<Object[]> parameterData = new ArrayList<Object[]>(Arrays.asList(new Object[][]
+                        {
+                                {MarshallingFormat.JAXB, jmsConfiguration} ,
+                                {MarshallingFormat.JSON, jmsConfiguration},
+                                {MarshallingFormat.XSTREAM, jmsConfiguration}
+                        }
+        ));
+
+        return parameterData;
+    }
+
+    @BeforeClass
+    public static void buildAndDeployArtifacts() {
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/definition-project").getFile());
+
+        createContainer(CONTAINER_ID, RELEASE_ID);
+    }
+
+    @After
+    public void resetResponseHandlers() {
```

Comment:
this needs to be removed as it was attempt to reset the handlers but it turned out it's not needed, thanks for pointing out.